src/Microsoft.Restier.EntityFramework/Model/ModelProducer.cs

```diff
@@ -336,11 +336,46 @@ private static IEdmPrimitiveTypeReference GetPrimitiveTypeReference(EdmProperty
                         efProperty.IsUnicode,
                         efProperty.Nullable);
                 case EdmPrimitiveTypeKind.DateTimeOffset:
-                case EdmPrimitiveTypeKind.Duration:
+                    if (efProperty.PrimitiveType.PrimitiveTypeKind == PrimitiveTypeKind.DateTime)
+                    {
+                        MetadataProperty metadata;
+                        efProperty.MetadataProperties.TryGetValue("Configuration", true, out metadata);
+                        if (metadata != null)
+                        {
+                            object config = metadata.Value;
+                            var columnType = (string)config.GetType().GetProperty("ColumnType").GetValue(config);
+
+                            if (string.Equals(columnType, "date", StringComparison.OrdinalIgnoreCase))
+                            {
+                                return EdmCoreModel.Instance.GetDate(efProperty.Nullable);
+                            }
+                        }
+                    }
```

Comment:
Is your code written specially for EF7 here? I am not sure about the difference that in my code the DB type name comes from the `ColumnAttribute`. But I guess we can keep both code here and use a compilation constant to differentiate EF6 and EF7?